### ISSUE TITLE
The difference between the paper and the implementation in R-MVSNet

### ISSUE THREAD

**Issue by tfj-dot** [YoYo000/MVSNet#112]
@YoYo000 
Hello. Let me ask a few questions to clarify on the differences between your R-MVSNet paper and the implementation of this repository.

I know that the following points were mentioned in Changelog

> Network change: enable scale and center in batch normalization
> Network change: replace UniNet with 2D UNet
> Network change: use group normalization in R-MVSNet

I saw that these changes are differences between your paper and the implementation.

Now, I have three questions.

1. The order of the convolutional layer and GRU in the cost regularization seems to be different between the paper and the implementation. **Was the cost regularization described in the paper used in the experiments?**

2. If the one in the paper was used, I am uncomfortable with the fact that Softmax's input is the output (-1 to 1) of the last GRU cell through tanh. This is because many classification networks do not apply activation to limit the value range before Softmax's input. **Did you apply output activation to the third GRU cell in the paper?**

3. On GRU, the paper discusses the nonlinear functions of R(t) and U(t)

> The nonlinear σg(-) is the hyperbolic tangent

It also says 

> σc(-) is the nonlinear mapping, which is the element-wise sigmoid function

But the implementation seems to be σg(-) is sigmoid, σc(-) is tanh. I think the implementation is a more general activation of GRU. Is the one in the paper a misdescription?

I'm sorry if I misread the paper and the code.
It would be helpful if you could reply.

**Comment by YoYo000** [comment]
@tfj-dot The model used in the R-MVSNet paper is exactly the same model implemented in this repository. 

Some minor modifications were made after we finish the paper writing. Sorry for making the implementation detail ambiguous.

**Comment by tfj-dot** [comment]
@YoYo000 
I understand.
Thank you for your prompt replying!
